controllers/gatling_controller.go

```diff
@@ -968,127 +972,115 @@ func (r *GatlingReconciler) isGatlingCompleted(gatling *gatlingv1alpha1.Gatling)
 	}
 	return provider
 }
 
 func (r *GatlingReconciler) getCloudStorageRegion(gatling *gatlingv1alpha1.Gatling) string {
 	region := ""
-	if &gatling.Spec.CloudStorageSpec != nil && gatling.Spec.CloudStorageSpec.Region != "" {
+	if gatling.Spec.CloudStorageSpec.Region != "" {
 		region = gatling.Spec.CloudStorageSpec.Region
 	}
 	return region
 }
 
 func (r *GatlingReconciler) getCloudStorageBucket(gatling *gatlingv1alpha1.Gatling) string {
 	bucket := ""
-	if &gatling.Spec.CloudStorageSpec != nil && gatling.Spec.CloudStorageSpec.Bucket != "" {
+	if gatling.Spec.CloudStorageSpec.Bucket != "" {
 		bucket = gatling.Spec.CloudStorageSpec.Bucket
 	}
 	return bucket
 }
 
 func (r *GatlingReconciler) getNotificationServiceProvider(gatling *gatlingv1alpha1.Gatling) string {
 	provider := defaultNotificationServiceProvider
-	if &gatling.Spec.NotificationServiceSpec != nil && gatling.Spec.NotificationServiceSpec.Provider != "" {
+	if gatling.Spec.NotificationServiceSpec.Provider != "" {
 		provider = gatling.Spec.NotificationServiceSpec.Provider
 	}
 	return provider
 }
 
 func (r *GatlingReconciler) getNotificationServiceSecretName(gatling *gatlingv1alpha1.Gatling) string {
 	secretName := ""
-	if &gatling.Spec.NotificationServiceSpec != nil && gatling.Spec.NotificationServiceSpec.SecretName != "" {
+	if gatling.Spec.NotificationServiceSpec.SecretName != "" {
 		secretName = gatling.Spec.NotificationServiceSpec.SecretName
 	}
 	return secretName
 }
 
 func (r *GatlingReconciler) getGatlingRunnerJobStartTime(gatling *gatlingv1alpha1.Gatling) string {
 	var startTime string
-	if &gatling.Spec.TestScenarioSpec != nil && gatling.Spec.TestScenarioSpec.StartTime != "" {
+	if gatling.Spec.TestScenarioSpec.StartTime != "" {
 		startTime = gatling.Spec.TestScenarioSpec.StartTime
 	}
 	return startTime
 }
 
 func (r *GatlingReconciler) getGatlingRunnerJobParallelism(gatling *gatlingv1alpha1.Gatling) *int32 {
 	var parallelism int32
-	if &gatling.Spec.TestScenarioSpec != nil && gatling.Spec.TestScenarioSpec.Parallelism != 0 {
+	if gatling.Spec.TestScenarioSpec.Parallelism != 0 {
 		parallelism = gatling.Spec.TestScenarioSpec.Parallelism
 	}
 	return &parallelism
 }
 
 func (r *GatlingReconciler) getGatlingContainerImage(gatling *gatlingv1alpha1.Gatling) string {
 	image := defaultGatlingImage
-	if &gatling.Spec.PodSpec != nil && gatling.Spec.PodSpec.GatlingImage != "" {
+	if gatling.Spec.PodSpec.GatlingImage != "" {
 		image = gatling.Spec.PodSpec.GatlingImage
 	}
 	return image
 }
 
 func (r *GatlingReconciler) getRcloneContainerImage(gatling *gatlingv1alpha1.Gatling) string {
 	image := defaultRcloneImage
-	if &gatling.Spec.PodSpec != nil && gatling.Spec.PodSpec.RcloneImage != "" {
+	if gatling.Spec.PodSpec.RcloneImage != "" {
 		image = gatling.Spec.PodSpec.RcloneImage
 	}
 	return image
 }
 
 func (r *GatlingReconciler) getPodResources(gatling *gatlingv1alpha1.Gatling) corev1.ResourceRequirements {
-	resources := corev1.ResourceRequirements{}
```

Comment:
Is this a breaking change, isn't it ?
This change means that If `gatling.Spec.PodSpec.Resources` is not nil, the value wouldn't be used.

Reply:
@gold-kou 
I believe that both &gatling.Spec.PodSpec and &gatling.Spec.PodSpec.Resources will never be nil.
This is because both PodSpec and Resources are not pointer types.
Therefore, the IF statement will always be true, and I don’t think the value assigned in this line will ever be used.